build.gradle

```diff
@@ -123,21 +140,92 @@ configure(subprojects.findAll {it.name != 'sda-commons-bom' && it.name != 'sda-c
  * - example modules and dependency-check should not be published at all
  */
 List<String> unpublishedModules = Arrays.asList("sda-commons-bom", "sda-commons-dependencies", "sda-commons-dependency-check")
-configure(subprojects.findAll {
-    !unpublishedModules.contains(it.name) && !it.name.endsWith("-example")
-}) {
+configure(subprojects.findAll { !unpublishedModules.contains(it.name) && !it.name.endsWith("-example") }) {
 
     task sourcesJar(type: Jar, dependsOn: classes) {
         archiveClassifier = 'sources'
         from sourceSets.main.allJava
     }
 
+    task javadocJar(type: Jar, dependsOn: javadoc) {
+        archiveClassifier = 'javadoc'
+        from javadoc.destinationDir
+    }
+
+    artifacts {
+        archives jar
+        archives sourcesJar
+        archives javadocJar
+    }
+
+    signing {
+        def signingKey = findProperty("signingKey")
+        def signingPassword = findProperty("signingPassword")
+        useInMemoryPgpKeys(signingKey, signingPassword)
+        sign configurations.archives
+    }
+
+    tasks.withType(Sign) {
+        onlyIf { gradle.taskGraph.hasTask(publishMavenPublicationToMavenCentralRepository) }
+    }
+
     publishing {
         publications {
             maven(MavenPublication) {
+                // create a custom "Project Object Model"
+                pom {
+                    name = project.group + ":" + project.name
+                    description = 'A libraries to bootstrap services easily that follow the patterns and specifications promoted by the SDA SE'
+                    url = 'https://github.com/SDA-SE/sda-dropwizard-commons'
+                    licenses {
+                        license {
+                            name = 'MIT License'
+                            url = 'https://raw.githubusercontent.com/SDA-SE/sda-dropwizard-commons/master/LICENSE'
+                        }
+                    }
+                    organization {
+                        name = 'SDA SE Open Industry Solutions'
+                        url = 'https://sdase.org'
```

Comment:
Do we have a constraint to use the same domain as our group, or can we use sda.se directly here?